Tests/tests/test_SystemServices.cpp

```diff
@@ -1281,3 +1285,514 @@ TEST_F(SystemServicesEventIarmTest, onRebootRequest)
+                 strncpy(pstParamData->value, expectedSerialNumber, sizeof(pstParamData->value));
+                 return WDMP_SUCCESS;
+            }));
+     EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getSerialNumber"), _T("{}"), response));
+     EXPECT_THAT(response, string("{\"serialNumber\":\"32E10400103240447\",\"success\":true}"));
+}
+
+#else
+/**
+ * @brief : getSerialNumber when getStateDetails.sh does not exist
+ *        Check if /lib/rdk/getStateDetails.sh file doesn't exist,
+ *        then getSerialNumber shall be failed and return the error message in response.
+ *
+ * @param[in]   :  This method takes no parameters.
+ * @return      :  {"SysSrv_Status":4,"errorMessage":"Unexpected Error","success":false}
+ */
+TEST_F(SystemServicesTest, getSerialNumberSnmpFailed_WhenScriptFileNotExist)
+{
+    const string deviceStateInfoScript = _T("/lib/rdk/getStateDetails.sh");
+    Core::File file(deviceStateInfoScript);
+    file.Create();
+    // Remove the file to simulate it is  not-existing
+    file.Destroy();
+    EXPECT_FALSE(Core::File(string(_T("/lib/rdk/getStateDetails.sh"))).Exists());
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("getSerialNumber"), _T("{}"), response));
+}
+
+/**
+ * @brief : getSerialNumber when TMP_SERIAL_NUMBER_FILE does not exist.
+ *        Check if TMP_SERIAL_NUMBER_FILE doesn't exist,then getSerialNumber shall be failed and
+ *        return the error message in response
+ *
+ * @param[in]   :  This method takes no parameters.
+ * @return      :  {"SysSrv_Status":4,"errorMessage":"Expected file not found","success":false}
+ */
+TEST_F(SystemServicesTest, getSerialNumberSnmpFailed_WhenTmpSerialNumberFileNotExist)
+{
+    const string deviceStateInfoScript = _T("/lib/rdk/getStateDetails.sh");
+    Core::File file(deviceStateInfoScript);
+    file.Create();
+    
+    ofstream file2("/tmp/.STB_SER_NO");
+    file2 << "32E10400103240447";
+    file2.close();
+    // Remove the file to simulate it is  not-existing
+    std::remove("/tmp/.STB_SER_NO");
+
+    EXPECT_TRUE(Core::File(string(_T("/lib/rdk/getStateDetails.sh"))).Exists());
+    EXPECT_CALL(wrapsImplMock, system(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](const char* command) {
+                EXPECT_EQ(string(command), string(_T("/lib/rdk/getStateDetails.sh STB_SER_NO")));
+                return 0;
+            }));
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("getSerialNumber"), _T("{}"), response));
+}
+
+/**
+ * @brief : getSerialNumber when TMP_SERIAL_NUMBER_FILE failed to read.
+ *        Check if contents of TMP_SERIAL_NUMBER_FILE can not be open,
+ *        then getSerialNumber shall be failed
+ *
+ * @param[in]   :  This method takes no parameters.
+ * @return      :  {"SysSrv_Status":6,"errorMessage":"Unsupported file content","success":false}
+ */
+TEST_F(SystemServicesTest, getSerialNumberSnmpFailed_WhenFailedToReadFromTmpFile)
```

Comment:
Are we planning to implement this test? If not pls remove.

Reply:
> Are we planning to implement this test? If not pls remove.

Yes Anand,we will work on it